scripts/sanitise-post-award-database.sh

```diff
@@ -0,0 +1,199 @@
+  echo "Waiting for 30 seconds..."
+  sleep 30;
+done
+
+# Connect to the bastion so that we can reach the DB
+BASTION_INSTANCE_ID=$(
+  aws-vault exec $AWS_VAULT_PROFILE_TO_SANITISE -- \
+  aws ec2 describe-instances \
+  --filters Name=tag:Name,Values="*-bastion" "Name=instance-state-name,Values='running'" \
+  --query "Reservations[*].Instances[*].InstanceId" \
+  | jq -r '.[0][0]'
+)
+
+REMOTE_SANITISE_DB_PORT=5432
+LOCAL_SANITISE_DB_PORT=1437
+DB_TO_SANITISE_HOST=$(echo $SANITISATION_DB | jq -r '.DBCluster.Endpoint')
+DB_TO_SANITISE_SECRET=$(aws-vault exec $AWS_VAULT_PROFILE_TO_SANITISE -- aws secretsmanager list-secrets | jq '.SecretList[] | select(.Tags[] | select(.Key == "aws:cloudformation:logical-id" and .Value == "postawardclusterAuroraSecret"))')
+DB_TO_SANITISE_PASSWORD=$(aws-vault exec $AWS_VAULT_PROFILE_TO_SANITISE -- aws secretsmanager get-secret-value --secret-id $(echo $DB_TO_SANITISE_SECRET | jq -r '.Name') | jq -r '.SecretString' | jq -r '.password')
+DB_TO_SANITISE_URI="postgresql://postgres:${DB_TO_SANITISE_PASSWORD}@localhost:${LOCAL_SANITISE_DB_PORT}/post_award"
+
+
+trap 'unset DB_TO_SANITISE_PASSWORD' EXIT INT TERM
+
+aws-vault exec $AWS_VAULT_PROFILE_TO_SANITISE -- \
+  aws ssm start-session \
+  --target $BASTION_INSTANCE_ID \
+  --document-name AWS-StartPortForwardingSessionToRemoteHost \
+  --parameters host="$DB_TO_SANITISE_HOST",portNumber="$REMOTE_SANITISE_DB_PORT",localPortNumber="$LOCAL_SANITISE_DB_PORT" &
+
+BASTION_SESSION_ID=$!
+sleep 10;
+
+echo "removing postgresql_anonymizer if already exist"
+rm -rf postgresql_anonymizer
+
+# Clone the repository and Install the postgres anonymiser
+echo "Cloning the repository..."
+git clone https://gitlab.com/dalibo/postgresql_anonymizer.git
+cd ./postgresql_anonymizer || { echo "Failed to enter repository directory"; exit 1; }
+
+# Checkout the version that allows standalone installation
+echo "Checking out version 0.8.1..."
+git checkout 0.8.1 || { echo "Failed to checkout version 0.8.1"; exit 1; }
+
+# Create the anon_standalone.sql file
+echo "Creating the anon_standalone.sql file..."
+make standalone || { echo "Failed to create anon_standalone.sql"; exit 1; }
+
+# Execute the SQL file against your Amazon RDS instance
+echo "Executing anon_standalone.sql against RDS instance..."
+psql $DB_TO_SANITISE_URI -f anon_standalone.sql || { echo "Failed to execute anon_standalone.sql"; exit 1; }
+
+# Navigate back to the original directory
+cd "$ORIGINAL_DIR" || { echo "Failed to return to the original directory"; exit 1; }
+echo "Anonymising db now"
+
+psql $DB_TO_SANITISE_URI -f sanitise-db.sql || { echo "Failed to execute anonmisation.sql"; exit 1; }
+echo "Process completed and anonymisation successfully."
+
+# Create backup
+pg_dump $DB_TO_SANITISE_URI -v -F c --clean --schema=public -f $BACKUP_FILE
+
+# Upload to S3 aws s3 ls
+aws-vault exec $AWS_VAULT_PROFILE_TO_SANITISE aws s3 cp $BACKUP_FILE s3://$S3_BUCKET/$S3_PATH
+
+trap clean EXIT INT TERM
```

Comment:
Again, this trap feels like it's in the wrong place - it won't cleanup the bastion session if any of the commands in between fail.

Happy to have a chat about this if it's not clear how to use traps